x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -1277,9 +1280,9 @@ protected LogicalPlan rule(Aggregate aggregate) {
                             Attribute attr = expToAttribute.computeIfAbsent(field.canonical(), k -> {
                                 Alias newAlias = new Alias(k.source(), syntheticName(k, af, counter[0]++), null, k, null, true);
                                 evals.add(newAlias);
-                                aggsChanged.set(true);
                                 return newAlias.toAttribute();
                             });
+                            aggsChanged.set(true);
```

Comment:
This caused the bug @astefan discovered - when the replacement was done for an expression twice (both inside the grouping AND the aggregates), the flag was not set (it was only triggered for expressions that were not seen before).
This caused the old aggregations to be used and thus the change to be ignored.
This issue existed before this PR which exposed it.